score-lang-compiler/src/main/java/org/openscore/lang/compiler/modeller/ExecutableBuilder.java

```diff
@@ -111,25 +111,34 @@ public Executable transformToExecutable(ParsedSlang parsedSlang, String execName
                 if(!executableRawData.containsKey(SlangTextualKeys.WORKFLOW_KEY)){
                     throw new RuntimeException("Error compiling " + parsedSlang.getName() + ". Flow: " + execName + " has no workflow property");
                 }
-                LinkedHashMap<String, Map<String, Object>> workFlowRawData;
+                List<Map<String, Map<String, Object>>> workFlowRawData;
                 try{
-                    workFlowRawData = (LinkedHashMap) executableRawData.get(SlangTextualKeys.WORKFLOW_KEY);
+                    workFlowRawData = (List) executableRawData.get(SlangTextualKeys.WORKFLOW_KEY);
                 } catch (ClassCastException ex){
-                    throw new RuntimeException("Flow: '" + execName + "' syntax is illegal.\nBelow 'workflow' property there should be a map of tasks and not a list");
+                    throw new RuntimeException("Flow: '" + execName + "' syntax is illegal.\nBelow 'workflow' property there should be a list of tasks and not a map");
                 }
-                if (MapUtils.isEmpty(workFlowRawData)) {
+                if (workFlowRawData == null || workFlowRawData.size() == 0) {
                     throw new RuntimeException("Error compiling " + parsedSlang.getName() + ". Flow: " + execName + " has no workflow data");
                 }
 
                 Workflow onFailureWorkFlow = null;
-                LinkedHashMap<String, Map<String, Object>> onFailureData;
-                try{
-                    onFailureData = (LinkedHashMap) workFlowRawData.remove(SlangTextualKeys.ON_FAILURE_KEY);
-                } catch (ClassCastException ex){
-                    throw new RuntimeException("Flow: '" + execName + "' syntax is illegal.\nBelow 'on_failure' property there should be a map of tasks and not a list");
-                }
-                if (MapUtils.isNotEmpty(onFailureData)) {
-                    onFailureWorkFlow = compileWorkFlow(onFailureData, imports, null, true);
+                List<Map<String, Map<String, Object>>> onFailureData;
+                Iterator<Map<String, Map<String, Object>>> tasksIterator = workFlowRawData.iterator();
+                while(tasksIterator.hasNext()){
+                    Map<String, Map<String, Object>> taskData = tasksIterator.next();
```

Comment:
What happens if onfailure is not the first task in the workflow order, how can another step know it should navigate to it instead of the default failure step? I think it has to happen before compiling the workflow this is whhy I wrote this ugly implementation before.